sonar-batch-protocol/src/main/java/org/sonar/batch/protocol/viewer/ViewerApplication.java

```diff
@@ -0,0 +1,341 @@
+      }
+    }
+  }
+
+  /**
+   * Create the application.
+   */
+  public ViewerApplication() {
+    initialize();
+  }
+
+  /**
+   * Initialize the contents of the frame.
+   */
+  private void initialize() {
+    try {
+      for (LookAndFeelInfo info : UIManager.getInstalledLookAndFeels()) {
+        if ("Nimbus".equals(info.getName())) {
+          UIManager.setLookAndFeel(info.getClassName());
+          break;
+        }
+      }
+    } catch (Exception e) {
+      // If Nimbus is not available, you can set the GUI to another look and feel.
+    }
+    frame = new JFrame();
+    frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+
+    splitPane = new JSplitPane();
+    frame.getContentPane().add(splitPane, BorderLayout.CENTER);
+
+    tabbedPane = new JTabbedPane(JTabbedPane.TOP);
+    tabbedPane.setPreferredSize(new Dimension(500, 7));
+    splitPane.setRightComponent(tabbedPane);
+
+    componentDetailsTab = new JScrollPane();
+    tabbedPane.addTab("Component details", null, componentDetailsTab, null);
+
+    componentEditor = new JEditorPane();
+    componentDetailsTab.setViewportView(componentEditor);
+
+    sourceTab = new JScrollPane();
+    tabbedPane.addTab("Source", null, sourceTab, null);
+
+    sourceEditor = createSourceEditor();
+    sourceEditor.setEditable(false);
+    sourceTab.setViewportView(sourceEditor);
+
+    textLineNumber = createTextLineNumber();
+    sourceTab.setRowHeaderView(textLineNumber);
+
+    highlightingTab = new JScrollPane();
+    tabbedPane.addTab("Highlighting", null, highlightingTab, null);
+
+    highlightingEditor = new JEditorPane();
+    highlightingTab.setViewportView(highlightingEditor);
+
+    coverageTab = new JScrollPane();
+    tabbedPane.addTab("Coverage", null, coverageTab, null);
+
+    coverageEditor = new JEditorPane();
+    coverageTab.setViewportView(coverageEditor);
+
+    treeScrollPane = new JScrollPane();
+    treeScrollPane.setPreferredSize(new Dimension(200, 400));
+    splitPane.setLeftComponent(treeScrollPane);
+
+    componentTree = new JTree();
+    componentTree.setModel(new DefaultTreeModel(
+      new DefaultMutableTreeNode("empty") {
+        {
+        }
+      }));
+    treeScrollPane.setViewportView(componentTree);
+    componentTree.getSelectionModel().setSelectionMode(TreeSelectionModel.SINGLE_TREE_SELECTION);
+    componentTree.addTreeSelectionListener(new TreeSelectionListener() {
+      public void valueChanged(TreeSelectionEvent e) {
+        DefaultMutableTreeNode node = (DefaultMutableTreeNode) componentTree.getLastSelectedPathComponent();
+
+        if (node == null) {
+          // Nothing is selected.
+          return;
+        }
+
+        updateDetails((Component) node.getUserObject());
+
+      }
+
+    });
+    frame.pack();
+  }
+
+  public JTree getComponentTree() {
+    return componentTree;
+  }
+
+  /**
+   * @wbp.factory
+   */
+  public static JPanel createComponentPanel() {
+    JPanel panel = new JPanel();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "panel". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1488)